### PR TITLE
重构自动刷新

### LINKED ISSUE #101: [optimize] banlog 页面增加加载动画
image
如图所示，现在banlog加载需要一定时间，需要增加加载动画以改善用户体验

### LINKED ISSUE #113: [todo] 在 PBH 启动之前打开 WebUI 将导致自动刷新失效，并阻止其它页面的后续所有网络请求
a27b1526de8f23ad623b32662e23e5e2

以前的版本WebUI打开的时候如果 PBH 没启动，在启动完成后页面可以恢复正常刷新
重构自动刷新之后，如果在加载 manifest 的时候失败了，好像就不会尝试恢复页面了

先打开 WebUI -> 再打开 PBH -> (出现问题) WebUI 停止加载，此时不管再怎么切换，也不会再发起任何新的加载请求。

不知道是不是连接卡住它就不加载其它请求了，还是一个请求失败了就不加载其它请求了。

### PR DESCRIPTION
**Pull request by XiaoXice**
接管vue-request自动刷新， 并统一实现。
可以手动触发刷新
加载动画对齐网络请求
also fix: #101 不过动画效果还需讨论，这个按钮上显示是不是有点太小了。
fix: #113 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new plugin system for managing auto-update functionality across various components.
	- Enhanced the `usePagination` function to support plugins for improved customization.
	- Added new localization support for network error messages, improving user guidance during connectivity issues.
	- Implemented a retry notification system for handling network failures effectively.

- **Improvements**
	- Improved user feedback in the auto-update button with enhanced loading state indicators.
	- Streamlined the component structure for better readability and maintainability.
	- Enhanced error handling logic in the settings modal for a better user experience.
	- Enhanced integration of notifications for a more dynamic user interaction.

- **Bug Fixes**
	- Addressed issues related to visibility and online status in the auto-update process.

- **Documentation**
	- Updated component import statements to reflect changes in the auto-update implementation.
	- Enhanced type safety with new utility types for timeouts.

### CODE REVIEW

**Comment by XiaoXice** [comment]
https://github.com/PBH-BTN/pbh-fe/blob/66409f10764e7597edc5aa7e829d4bc0ce1d5318/src/components/pageHeader.vue#L99
todo: 需要把这个多余的参数干掉

File: package.json [hunk 1]
```diff
@@ -35,31 +35,36 @@
...
+    "vite-plugin-vue-devtools": "^7.3.7",
     "vue-tsc": "^2.0.29"
   },
   "engines": {
     "node": ">=20.0.0",
     "pnpm": ">=9.0.0"
+  },
+  "pnpm": {
+    "patchedDependencies": {
+      "vue-request@2.0.4": "patches/vue-request@2.0.4.patch"
```
Comment (Gaojianli) [inline package.json:72]: 这是什么黑科技？？？

File: src/views/banlist/components/banList.vue [hunk 2]
```diff
@@ -2,14 +2,27 @@
...
-    }" ref="banlist" @reach-bottom="loadMore" :scrollbar="false" :data="list">
+    <a-list
+      :virtualListProps="{
+        height: virtualListHeight
+      }"
+      ref="banlist"
+      @reach-bottom="loadMore"
+      :scrollbar="false"
+      :data="list"
+    >
```
Comment (Gaojianli) [inline src/views/banlist/components/banList.vue:21]: 没看到spin，怎么做的 #101

**Review by Gaojianli — CHANGES_REQUESTED** [review]

File: src/views/banlist/components/banList.vue (hunk 2 above)
Comment (XiaoXice) [reply src/views/banlist/components/banList.vue:21]: 现在加上了

File: package.json (hunk 1 above)
Comment (XiaoXice) [reply package.json:72]: like to: https://github.com/AttoJS/vue-request/pull/248

**Review by Gaojianli — APPROVED** [review]

**Pull request merged**